packages/cosmos_ui_components/lib/components/cosmos_text_field.dart

```diff
@@ -42,17 +49,33 @@ class _CosmosTextFieldState extends State<CosmosTextField> {
       controller: controller,
       maxLines: widget.maxLines,
       maxLength: widget.maxLength,
-      onChanged: widget.onChanged,
+      keyboardType: widget.keyboardType,
+      onChanged: (value) {
+        widget.onChanged(value);
+        setState(() {});
+      },
       decoration: InputDecoration(
         border: UnderlineInputBorder(borderSide: BorderSide(color: theme.colors.inputBorder)),
+        hintText: widget.hint,
+        hintStyle: CosmosTextTheme.copy0Normal,
         counterText: '',
-        suffix: InkWell(
-          onTap: () {
-            controller.clear();
-            widget.onChanged('');
-          },
-          child: SizedBox(height: 17, width: 17, child: Image.asset('assets/images/cross.png', package: packageName)),
-        ),
+        suffixIcon: controller.text.isNotEmpty ? null : widget.suffix,
+        suffix: widget.suffix != null ? (controller.text.isNotEmpty ? _buildClearButton() : null) : _buildClearButton(),
+      ),
+    );
+  }
+
+  InkWell _buildClearButton() {
+    return InkWell(
+      onTap: () {
+        controller.clear();
+        widget.onChanged('');
```

Comment:
you already call `onChanged` on each character change, so no need to call it again here explicitly

Reply:
Updated